src/test/scala/cognite/spark/v1/DataModelInstancesRelationTest.scala

```diff
@@ -198,6 +198,22 @@ class DataModelInstancesRelationTest
     )
   }
 
+  it should "return an informative error when a value with wrong type is attempted to be ingested" in {
+    val ex = sparkIntercept {
+      insertRows(
+        primitiveExtId,
+        spark
+          .sql(s"""select '2.0' as prop_float,
+                  |true as prop_bool,
+                  |'abc' as prop_string,
+                  |'test' as externalId""".stripMargin)
+      )
+    }
+    ex shouldBe an[CdfSparkException]
+    ex.getMessage shouldBe "2.0 of type class java.lang.String is not a valid float64. " +
+      "Try to cast the value to double. For example, ‘double(col_name) as prop_name’ or ‘cast(col_name as double) as prop_name’."
```

Comment:
Maybe you can also add 2 more tests with query : `select cast('2.0' as double) as prop_float` and `select double('2.0')  as prop_float`to make sure the casting work 😁

Reply:
aha good idea, let me do it

Reply:
can you check?

Reply:
😸 ✅